src/edge/simulator.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+import * as dotenv from "dotenv";
+import * as fse from "fs-extra";
+import * as os from "os";
+import * as path from "path";
+import * as request from "request-promise";
+import * as semver from "semver";
+import * as vscode from "vscode";
+import { Constants } from "../common/constants";
+import { Executor } from "../common/executor";
+import { LearnMoreError } from "../common/LearnMoreError";
+import { UserCancelledError } from "../common/UserCancelledError";
+import { Utility } from "../common/utility";
+import { IDeviceItem } from "../typings/IDeviceItem";
+
+enum InstallReturn {
+    Success = 0,
+    Failed,
+    Canceled,
+    NoPip,
+}
+
+export class Simulator {
+    public static async validateSimulatorUpdated(outputChannel: vscode.OutputChannel = null): Promise<void> {
+        let message: string;
+        try {
+            const output = await Executor.executeCMD(undefined, "iotedgehubdev", { shell: true }, "--version");
```

Comment:
The iotedgehubdev command here (and below) runs in output channel, so users' virtual env won't work.

Reply:
I would like fix virtual env as an issue. Does that mean virtual env is not setup in output? only for terminal?

Reply:
Virtualenv only applies to terminals.

Reply:
will update install page to ask user install iotedgehubdev in global python env. New feature request may be specify the iotedgehubdev path of virtual env.